internal/app/proxyd/internal/frontend/frontend.go

```diff
@@ -86,24 +74,34 @@ func (r *ReverseProxy) Listen(address string) (err error) {
 func (r *ReverseProxy) AddBackend(uid, addr string) (added bool) {
 	r.mux.Lock()
 	defer r.mux.Unlock()
+
+	// Check if we already have this backend added
 	if _, ok := r.backends[uid]; ok {
-		added = false
-		return
+		if r.backends[uid].Addr == addr {
```

Comment:
I don't understand what this check buys us. If `r.backends[uid].Addr != addr`, what does that mean? Just trying to follow where you were going with this and why, so I can understand it.

Reply:
It's more of an additional safety measure since I'm not certain the behavior when a pod gets updated.
I'm assuming during a pod update it would generate a new uuid so this probably isn't adding much value. As I was thinking through it, it seemed like a sane additional check to add if for whatever reason a pod IP changes but the uid is the same.